### PR TITLE
goreleaser: add brew formula and fix ldflags

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

@chmouel I may need access to your brew tap repository :angel: 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ :no_good_man: ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ :no_good_man: ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -45,3 +45,16 @@ changelog:
 release:
   draft: true
   prerelease: auto
+brews:
+  -
+    name: tektoncd-cli
+    github:
+      owner: chmouel
+      name: homebrew-tektoncd-cli
```
Comment (chmouel) [inline .goreleaser.yml]: do you think we can create an `official` one on tektoncd ?

File: .goreleaser.yml [hunk 2]
```diff
@@ -45,3 +45,16 @@ changelog:
...
+    folder: Formula
+    homepage: "https://github.com/tektoncd/cli"
+    description: A CLI for interacting with Tekton!
+    test: |
+      system "#{bin}/tkn --version"
+    install: |
```
Comment (chmouel) [inline .goreleaser.yml]: according to homebrew code, this should be implicit fyi : https://github.com/goreleaser/goreleaser/blob/3f082c9252eb93a4bdf5045e06accbde37bb6dc3/internal/pipe/brew/brew.go#L66-L80

**Comment by chmouel** [comment]
I think it may be better to have a official repo on `tektoncd` since migration could get annoying if people starts to use mine,

File: .goreleaser.yml (hunk 2 above)
Comment (vdemeester) [reply .goreleaser.yml]: oh if I can remove it, I will !

**Pull request merged**